### PR TITLE
Add table aws_tagging_resource closes #618

### LINKED ISSUE #618: Add table aws_tagging_resource
**References**
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/resourcegroupstaggingapi_resources
https://docs.aws.amazon.com/sdk-for-go/api/service/resourcegroupstaggingapi/

Terraform has the full service name, `resourcegroupstaggingapi`, but to make it a bit more readable, we should go with `tagging`.

This table should support optional key quals for filters and resource ARN lists.

### PR DESCRIPTION
**Pull request by ParthaI**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
No env file present for the current environment:  staging
customEnv TURBOT_TEST_EXPECTED_TIMEOUT undefined

SETUP: tests/aws_tagging_resource []

PRETEST: tests/aws_tagging_resource

TEST: tests/aws_tagging_resource
Running terraform
data.aws_region.alternate: Refreshing state...
data.aws_partition.current: Refreshing state...
data.aws_region.primary: Refreshing state...
data.aws_caller_identity.current: Refreshing state...
data.null_data_source.resource: Refreshing state...
aws_ebs_volume.named_test_resource: Creating...
aws_ebs_volume.named_test_resource: Still creating... [10s elapsed]
aws_ebs_volume.named_test_resource: Creation complete after 16s [id=vol-06eac213685267060]
data.aws_resourcegroupstaggingapi_resources.named_test_resource: Refreshing state...

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

Outputs:

account_id = 632902152528
aws_partition = aws
aws_region = us-east-1
resource_aka = arn:aws:ec2:us-east-1:632902152528:volume/vol-06eac213685267060
resource_id = vol-06eac213685267060
resource_name = turbottest70233
resource_tags = {
  "Name" = "turbottest70233"
}
resource_title = volume/vol-06eac213685267060

Running SQL query: test-get-query.sql
[
  {
    "account_id": "632902152528",
    "akas": [
      "arn:aws:ec2:us-east-1:632902152528:volume/vol-06eac213685267060"
    ],
    "arn": "arn:aws:ec2:us-east-1:632902152528:volume/vol-06eac213685267060",
    "name": "volume/vol-06eac213685267060",
    "partition": "aws",
    "region": "us-east-1",
    "tags": {
      "Name": "turbottest70233"
    }
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "arn": "arn:aws:ec2:us-east-1:632902152528:volume/vol-06eac213685267060",
    "name": "volume/vol-06eac213685267060"
  }
]
✔ PASSED

Running SQL query: test-notfound-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "account_id": "632902152528",
    "akas": [
      "arn:aws:ec2:us-east-1:632902152528:volume/vol-06eac213685267060"
    ],
    "region": "us-east-1",
    "tags": {
      "Name": "turbottest70233"
    },
    "title": "volume/vol-06eac213685267060"
  }
]
✔ PASSED

POSTTEST: tests/aws_tagging_resource

TEARDOWN: tests/aws_tagging_resource

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>

### Basic Info  

```sql
select
  name,
  arn,
  compliance_status,
  tags,
  region
from
  aws_tagging_resource;
```

```
+-------------------------------------------+---------------------------------------------------------------------------------------+-------------------+---------------------------------------------------
| name                                      | arn                                                                                   | compliance_status | tags                                              
+-------------------------------------------+---------------------------------------------------------------------------------------+-------------------+---------------------------------------------------
| rule/turbot_aws_api_events_rds            | arn:aws:events:ap-south-1:632902152528:rule/turbot_aws_api_events_rds                 | <null>            | {}                                                
| snapshot/snap-009f2ffae82714cab           | arn:aws:ec2:ap-south-1:632902152528:snapshot/snap-009f2ffae82714cab                   | <null>            | {"Name":"vol-02776515d142986a3@2020-11-30T03:31:28
| subnet/subnet-0989d6dd12c4ecf17           | arn:aws:ec2:us-east-1:632902152528:subnet/subnet-0989d6dd12c4ecf17                    | <null>            | {"Name":"test-01"}                                
| rule/turbot_aws_api_events_kms            | arn:aws:events:us-east-1:632902152528:rule/turbot_aws_api_events_kms                  | <null>            | {}                                                
| rule/turbot_aws_api_events_billingconsole | arn:aws:events:us-east-1:632902152528:rule/turbot_aws_api_events_billingconsole       | <null>            | {}                                                
| security-group/sg-05a5f107ab15d88d5       | arn:aws:ec2:ap-south-1:632902152528:security-group/sg-05a5f107ab15d88d5               | <null>            | {"test":"01"}                                     
| vpc-flow-log/fl-0c09d1baa285643a8         | arn:aws:ec2:ap-south-1:632902152528:vpc-flow-log/fl-0c09d1baa285643a8                 | <null>            | {"Name":"flo-log-test002"}                        
| rule/turbot_aws_api_events_ecs            | arn:aws:events:ap-south-1:632902152528:rule/turbot_aws_api_events_ecs                 | <null>            | {}                                                
| capacity-reservation/cr-0401ebd31315c8cc6 | arn:aws:ec2:us-east-1:632902152528:capacity-reservation/cr-0401ebd31315c8cc6          | <null>            | {"test-tag-key":"test-tag-value"}                 
| rule/turbot_aws_api_events_logs           | arn:aws:events:ap-south-1:632902152528:rule/turbot_aws_api_events_logs                | <null>            | {}                                                
| rule/turbot_aws_api_events_lambda         | arn:aws:events:us-east-1:632902152528:rule/turbot_aws_api_events_lambda               | <null>            | {}                                                
| mybucket307                               | arn:aws:s3:::mybucket307                                                              | <null>            | {"tag1":"val1","tag2":"val2"}                     
| rule/turbot_aws_api_events_iam            | arn:aws:events:ap-south-1:632902152528:rule/turbot_aws_api_events_iam                 | <null>            | {}                                                
| rule/turbot_aws_api_events_lambda         | arn:aws:events:ap-south-1:632902152528:rule/turbot_aws_api_events_lambda              | <null>            | {}                                                
| rule/turbot_aws_api_events_billingconsole | arn:aws:events:ap-south-1:632902152528:rule/turbot_aws_api_events_billingconsole      | <null>            | {}                                                
| alarm-action-active                       | arn:aws:cloudwatch:ap-south-1:632902152528:alarm:alarm-action-active                  | <null>            | {}                                                
| rule/turbot_aws_api_events_ssm            | arn:aws:events:ap-south-1:632902152528:rule/turbot_aws_api_events_ssm                 | <null>            | {}                                                
| rule/turbot_aws_api_events_cloudtrail     | arn:aws:events:ap-south-1:632902152528:rule/turbot_aws_api_events_cloudtrail          | <null>            | {}                                                
| rule/turbot_aws_api_events_iam            | arn:aws:events:us-east-1:632902152528:rule/turbot_aws_api_events_iam                  | <null>            | {}                                                
| rule/turbot_aws_api_events_s3             | arn:aws:events:us-east-1:632902152528:rule/turbot_aws_api_events_s3                   | <null>            | {}                                                
| image/ami-04c029923f36c6877               | arn:aws:ec2:us-east-1:632902152528:image/ami-04c029923f36c6877                        | <null>            | {"tag16":"val16","tag2":"val2","template":"turbot1
| arnab-hello-world-python                  | arn:aws:lambda:us-east-1:632902152528:function:arnab-hello-world-python               | <null>            | {"lambda-console:blueprint":"hello-world-python"} 
| rule/turbot_aws_api_events_sns            | arn:aws:events:ap-south-1:632902152528:rule/turbot_aws_api_events_sns                 | <null>            | {}                                                
| rule/xcxczxczxc                           | arn:aws:events:us-east-1:632902152528:rule/xcxczxczxc                                 | <null>            | {}                                                
| rule/9cd236e9-52b8-4570-ae67-6d6b274ce817 | arn:aws:waf-regional:us-east-1:632902152528:rule/9cd236e9-52b8-4570-ae67-6d6b274ce817 | <null>            | {}                                                
| rule/turbot_aws_api_events_rds            | arn:aws:events:us-east-1:632902152528:rule/turbot_aws_api_events_rds                  | <null>            | {}                                                
| rule/turbot_aws_api_events                | arn:aws:events:us-east-1:632902152528:rule/turbot_aws_api_events                      | <null>            | {}                                                
| rule/turbot_aws_api_events_ssm            | arn:aws:events:us-east-1:632902152528:rule/turbot_aws_api_events_ssm                  | <null>            | {}                                                
| vpc-flow-log/fl-0a8d853f5ecf38815         | arn:aws:ec2:ap-south-1:632902152528:vpc-flow-log/fl-0a8d853f5ecf38815                 | <null>            | {"Name":"second-flowlog-test"}                    
| rule/turbot_aws_api_events_ecs            | arn:aws:events:us-east-1:632902152528:rule/turbot_aws_api_events_ecs                  | <null>            | {}                                                
| policy/turbot/owner                       | arn:aws:iam::632902152528:policy/turbot/owner                                         | <null>            | {}                                                
| rule/turbot_aws_api_events_cloudtrail     | arn:aws:events:us-east-1:632902152528:rule/turbot_aws_api_events_cloudtrail           | <null>            | {}                                                
| rule/turbot_aws_api_events_config         | arn:aws:events:ap-south-1:632902152528:rule/turbot_aws_api_events_config              | <null>            | {}                                                
| vpc/vpc-05aa0d2485a8b16b1                 | arn:aws:ec2:us-east-1:632902152528:vpc/vpc-05aa0d2485a8b16b1                          | <null>            | {"Name":"test"}                                   
| rule/turbot_aws_api_events_sns            | arn:aws:events:us-east-1:632902152528:rule/turbot_aws_api_events_sns                  | <null>            | {}                                                
| rule/turbot_aws_api_events_s3             | arn:aws:events:ap-south-1:632902152528:rule/turbot_aws_api_events_s3                  | <null>            | {}                                                
| billing_greater_than_thresold_value       | arn:aws:cloudwatch:us-east-1:632902152528:alarm:billing_greater_than_thresold_value   | <null>            | {}                                                
| rule/turbot_aws_api_events_logs           | arn:aws:events:us-east-1:632902152528:rule/turbot_aws_api_events_logs                 | <null>            | {}                                                
| rule/turbot_aws_api_events_cloudformation | arn:aws:events:us-east-1:632902152528:rule/turbot_aws_api_events_cloudformation       | <null>            | {}                                                
| rule/turbot_aws_api_events_rds            | arn:aws:events:us-east-2:632902152528:rule/turbot_aws_api_events_rds                  | <null>            | {}                                                
| key/b4852ca5-859c-464c-9de1-c37e38352656  | arn:aws:kms:us-east-1:632902152528:key/b4852ca5-859c-464c-9de1-c37e38352656           | <null>            | {"name":"turbottest51410"}                        
| rule/turbot_aws_api_events_config         | arn:aws:events:us-east-1:632902152528:rule/turbot_aws_api_events_config               | <null>            | {}                                                
| turbot-bucket-version                     | arn:aws:s3:::turbot-bucket-version                                                    | <null>            | {"Company":"Vandelay Industries","Cost Center":"31
| rule/turbot_aws_api_events_sns            | arn:aws:events:us-east-2:632902152528:rule/turbot_aws_api_events_sns                  | <null>            | {}                                                
| rule/turbot_aws_api_events_config         | arn:aws:events:us-east-2:632902152528:rule/turbot_aws_api_events_config               | <null>            | {}                                                
| rule/turbot_aws_api_events_cloudformation | arn:aws:events:us-east-2:632902152528:rule/turbot_aws_api_events_cloudformation       | <null>            | {}                                                
| rule/turbot_aws_api_events                | arn:aws:events:us-east-2:632902152528:rule/turbot_aws_api_events                      | <null>            | {}                                                
| rule/turbot_aws_api_events_ecs            | arn:aws:events:us-east-2:632902152528:rule/turbot_aws_api_events_ecs                  | <null>            | {}                                                
| rule/turbot_aws_api_events_cloudtrail     | arn:aws:events:us-east-2:632902152528:rule/turbot_aws_api_events_cloudtrail           | <null>            | {}                                                
| volume/vol-03b8b46b5937f2c6e              | arn:aws:ec2:us-east-2:632902152528:volume/vol-03b8b46b5937f2c6e                       | <null>            | {"Volumesize":"2222"}                             
| rule/turbot_aws_api_events_logs           | arn:aws:events:us-east-2:632902152528:rule/turbot_aws_api_events_logs                 | <null>            | {}                                                
| rule/turbot_aws_api_events_lambda         | arn:aws:events:us-east-2:632902152528:rule/turbot_aws_api_events_lambda               | <null>            | {}                                                
| rule/turbot_aws_api_events_iam            | arn:aws:events:us-east-2:632902152528:rule/turbot_aws_api_events_iam                  | <null>            | {}                                                
| rule/turbot_aws_api_events_s3             | arn:aws:events:us-east-2:632902152528:rule/turbot_aws_api_events_s3                   | <null>            | {}                                                
| rule/turbot_aws_api_events_kms            | arn:aws:events:us-east-2:632902152528:rule/turbot_aws_api_events_kms                  | <null>            | {}                                                
| rule/turbot_aws_api_events_billingconsole | arn:aws:events:us-east-2:632902152528:rule/turbot_aws_api_events_billingconsole       | <null>            | {}                                                
| rule/turbot_aws_api_events_ssm            | arn:aws:events:us-east-2:632902152528:rule/turbot_aws_api_events_ssm                  | <null>            | {}                                                
:

```

```sql
select name, arn, tags, title from aws_tagging_resource where arn = 'arn:aws:events:ap-south-1:632902152528:rule/turbot_aws_api_events_iam';
```

```
+--------------------------------+-----------------------------------------------------------------------+------+--------------------------------+
| name                           | arn                                                                   | tags | title                          |
+--------------------------------+-----------------------------------------------------------------------+------+--------------------------------+
| rule/turbot_aws_api_events_iam | arn:aws:events:ap-south-1:632902152528:rule/turbot_aws_api_events_iam | {}   | rule/turbot_aws_api_events_iam |
+--------------------------------+-----------------------------------------------------------------------+------+--------------------------------+
```

</details>

### CODE REVIEW

File: aws/service.go [hunk 1]
```diff
@@ -1425,6 +1426,29 @@ func StsService(ctx context.Context, d *plugin.QueryData) (*sts.STS, error) {
 	return svc, nil
 }
 
+// TaggignResourceService returns the service connection for AWS ResourceTaggingAPI service
+func TaggignResourceService(ctx context.Context, d *plugin.QueryData) (*resourcegroupstaggingapi.ResourceGroupsTaggingAPI, error) {
+	region := d.KeyColumnQualString(matrixKeyRegion)
+	if region == "" {
+		return nil, fmt.Errorf("region must be passed SsmService")
```
Comment (bigdatasourav) [inline aws/service.go:1433]:
```suggestion
		return nil, fmt.Errorf("region must be passed TaggignResourceService")
```

File: aws/table_aws_tagging_resource.go [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+				Name:        "tags",
+				Description: resourceInterfaceDescription("tags"),
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("Tags").Transform(resourceTagListToTurbotTags),
+			},
+			{
+				Name:        "akas",
+				Description: resourceInterfaceDescription("akas"),
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("ResourceARN").Transform(arnToAkas),
```
Comment (bigdatasourav) [inline aws/table_aws_tagging_resource.go:80]:
```suggestion
				Transform:   transform.FromField("ResourceARN").Transform(transform.EnsureStringArray),
```

File: docs/tables/aws_tagging_resource.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
+# Table: aws_tagging_resource
+
+A patch baseline defines which patches are approved for installation on your instances.
```
Comment (bigdatasourav) [inline docs/tables/aws_tagging_resource.md:3]: Please provide proper description.

File: docs/tables/aws_tagging_resource.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  name,
+  arn,
+  compliance_status,
+  tags,
+  region
+from
+  aws_tagging_resource;
+```
+
+### List of resources where resource is compliant with the effective tag policy
```
Comment (bigdatasourav) [inline docs/tables/aws_tagging_resource.md:20]:
```suggestion
### List compliant resources
```

**Review by bigdatasourav — APPROVED** [review]
LGTM

**Review by rajlearner17 — APPROVED** [review]
LGTM

**Pull request merged**